drivers/adc/Kconfig.mcp356x

```diff
@@ -0,0 +1,74 @@
+# MCP356x ADC configuration options
+
+# Copyright (c) 2022 Trent Piepho <trent.piepho@igorinstitute.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config ADC_MCP356X
+	bool "MCP3561/2/4 driver"
+	depends on SPI
```

Comment:
Please update this to use the `depends on DT_HAS_...` pattern.

Reply:
That didn't exist when I submitted this PR.

It's not clear to me how the existence of a dt node compatible with this driver insures that SPI support is enabled.  What prevents a link failure if SPI is not enabled?

If this were Linux, I'd go look at the mailing list discussion of this new design to see how this is supposed to work.  But there's no public list for Zephyr.  I looked at the PRs and found #47196.  But real discussion is private and offline.  Doesn't really help.